python/lsst/ts/hvac/hvac_csc.py

```diff
@@ -183,6 +191,17 @@ def __init__(
         # and this gets initialized in the connect method.
         self.hvac_state = None
 
+        # Keep track of whether a deviceEnabled log event needs to be sent
+        self.send_deviceEnabled_event = True
```

Comment:
Are you sure you need this?

`salobj.ControllerEvent.set_put` only outputs an even *if* some field has changed (or if you specify force_output=True) and that's usually sufficient.

To put it another way: you should alway call set_put on an event if the event's data might have changed. It can then figure out whether the data actually changed and output the event if so. That is usually all you have to do in order to reliably report the current state of the event.

(Also if you want to know if the data was output: set_put returns a boolean did_put; I often use this for temporary diagnostic output when debugging -- I print the data of interest when it changes)

Reply:
I was not aware of that behavior so I removed the boolean. Thanks!